### PR TITLE
Add new Azure job for checking style

### LINKED ISSUE #102: Create a new azure pipelines that runs "make check"
Axom hopefully soon will create a new set of docker images (we both need a clang 10 one).  This will allow us to run ClangFormat to ensure the code style is enforced.

### PR DESCRIPTION
**Pull request by joshessman-llnl**
Now the Azure pipelines [fail](https://dev.azure.com/llnl-serac/serac/_build/results?buildId=581&view=logs&j=cb13b0c9-f406-547a-d755-9fe006f85343&t=435ca3cd-a81f-530c-87da-e79a43d1b3d6&l=241) if there are style deviations.

This required a minor change to the Dockerfiles/images due to https://github.com/LLNL/blt/issues/384

Closes #102

### CODE REVIEW

File: scripts/azure-pipelines/linux-check.sh [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+or_die cd serac
+git submodule init 
+git submodule update 
+
+echo HOST_CONFIG
+echo $HOST_CONFIG
+
+echo "~~~~~~ RUNNING CMAKE ~~~~~~~~"
+or_die ./config-build.py -hc /home/serac/serac/host-configs/docker/${HOST_CONFIG}.cmake -DENABLE_GTEST_DEATH_TESTS=ON
```
Comment (white238) [inline scripts/azure-pipelines/linux-check.sh]: Why was the death test option given? make check doesn't run gtest.

**Review by white238 — APPROVED** [review]

**Comment by white238** [comment]
I just gave you access to the BLT repo so feel free, or not, to submit a PR fixing the offending python script I pulled off the internet to work around clang format not having a dry run option.  It is supposed to be added soonish or is in a higher version than we can reliably support.

File: scripts/azure-pipelines/linux-check.sh (hunk 1 above)
Comment (joshessman-llnl) [reply scripts/azure-pipelines/linux-check.sh]: It was copy-pasted, will remove

**Review by jamiebramwell — APPROVED** [review]

**Comment by joshessman-llnl** [comment]
LGTM

**Pull request merged**